subprojects/language-java/src/main/java/org/gradle/api/internal/tasks/compile/AbstractJavaCompileSpecFactory.java

```diff
@@ -51,20 +52,32 @@ public T create() {
             }
 
             return getForkingSpec(Jvm.current().getJavaHome());
-        } else {
-            return getDefaultSpec();
         }
+
+        return getDefaultSpec();
     }
 
     private T chooseSpecForToolchain() {
         File toolchainJavaHome = toolchain.getInstallationPath().getAsFile();
         if (!toolchain.getLanguageVersion().canCompileOrRun(8)) {
             return getCommandLineSpec(Jvm.forHome(toolchainJavaHome).getJavacExecutable());
         }
-        if (!compileOptions.isFork() && toolchain.isCurrentJvm()) {
-            return getDefaultSpec();
+
+        if (compileOptions.isFork()) {
```

Comment:
I think it would be worth adding a section to the docs about toolchain priority and linking to it from javadoc on this method.